chopper_generator/lib/src/generator.dart

```diff
@@ -466,10 +488,41 @@ final class ChopperGenerator
     });
   }
 
+  static void _cleanMapBody(List<Code> blocks, String refName,
+      {bool valueToString = false, DartType? type}) {
+    if (type != null) {
+      final mapTypeChecker = TypeChecker.fromRuntime(Map);
+      if (!mapTypeChecker.isExactlyType(type)) {
+        return;
+      }
+    }
+    final mapType = type?.toString() ?? 'Map<String, dynamic>';
+    final idNullableValueType =
+        mapType.endsWith('dynamic>') || mapType.endsWith('?>');
+    final isStringValueType = mapType.endsWith('String>');
+    if (!idNullableValueType && isStringValueType) {
```

Comment:
Not sure if this is the nicest way of checking the types. It's possible to check inner type as far I know. See [Response type checker](https://github.com/lejard-h/chopper/blob/2249102880d84f21854475bd641f4d5ce336f0c5/chopper_generator/lib/src/generator.dart#L602).

 But doesn't `mapType.endsWith('String>')` also mean its non-nullable. So the if statement only needs to check if its a `isStringValueType`.